aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -84,72 +85,17 @@ public boolean hasPendingMutation(@NonNull String modelId) {
         if (existingMutation == null) {
```

Comment:
If we have a bunch of conditional updates come in, then don't we need to enqueue them all? So we would end up with a collection of mutations per model ID?